ext/cl_khr_layers.asciidoc

```diff
@@ -0,0 +1,182 @@
+// Copyright 2017-2020 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+[[cl_khr_layers-opencl]]
+== Installable OpenCL Loader Layers
+
+[[cl_khr_layers-overview]]
+=== Overview
+
+This section describes *cl_khr_layers* loader extension which defines
+a simple mechanism through which the Khronos OpenCL installable client
+driver loader (ICD Loader) may expose load multiple separate installable
+interception layers (Layer ICDs) for OpenCL.
+
+=== General information
+
+==== Version history
+
+[cols="1,1,3",options="header",]
+|====
+| *Date*     | *Version* | *Description*
+| 2020-11-04 | 1.0.0     | First assigned version.
+|====
+
+[[cl_khr_layers-new-procedures-and-functions]]
+=== New Procedures and Functions
+
+A layer needs to implement and expose those two new entry points in
+a shared library. If one or both of those are missing, the loader will
+discard the layer.
+
+[source,c]
+----
+cl_int clGetLayerInfo(cl_layer_info  param_name,
+                      size_t         param_value_size,
+                      void          *param_value,
+                      size_t        *param_value_size_ret);
+
+cl_int clInitLayer(cl_uint                 num_entries,
+                   const cl_icd_dispatch  *target_dispatch,
+                   cl_uint                *num_entries_ret,
+                   const cl_icd_dispatch **layer_dispatch);
+----
+
+[[cl_khr_layers-new-api-types]]
+=== New API Types
+
+[source]
+----
+typedef cl_uint cl_layer_info;
+typedef cl_uint cl_layer_api_version;
```

Comment:
Not sure technical changes are appropriate now, but should this typedef be removed and instances of `cl_layer_api_version` be replaced with `cl_version`?

It wasn't added until OpenCL 3.0, but as layers only work with the latest ICD loader and headers I'm not sure that matters.

Then values would be produced via `CL_MAKE_VERSION_KHR(1, 0, 0)`, which would produce a different number than 100.

I don't feel strongly about this, especially given the code already merged into the ICD loader.  It is completely fine to resolve this comment and leave things as is if you think that's best.

Reply:
I don't have a strong opinion on the subject. This is confined to the loader, and I don't expect the API to incrementally change much, so the simple API mechanism should suffice, but if the majority want to change it I wont fight it.